rest_framework/compat.py

```diff
@@ -157,3 +158,19 @@ def md_filter_add_syntax_highlight(md):
 SHORT_SEPARATORS = (',', ':')
 LONG_SEPARATORS = (', ', ': ')
 INDENT_SEPARATORS = (',', ': ')
+
+
+if django.VERSION > (4, 1):
+    from django.utils.http import parse_header_parameters
+
+    def parse_header_params(params, encoding='utf-8'):
+        key, pdict =  parse_header_parameters(params)
+        # parse_header_params expects values in the bytes format and
+        # returns string values.
+        pdict = {k:v.encode(encoding) for k, v in pdict.items()}
+        return key, pdict
```

Comment:
Can we adjust the client code (the calling code) to expect the new format, and then adjust the fallback to map to that? 🤔

The ideal is that when Django 4.2 is the minimum supported version we can delete the code in `compat`, change the imports elsewhere and be done.

Reply:
originally, i too was aiming in that direction. but since the returned value type are different from the two functions, and based upon that the client does its manipulation, this seemed to be the way with minimal change.

if we want to adjust the client code everywhere, it will have to change everywhere depending upon the version or return value type. to keep the logic at one place right now, kept it inside `compat.py`.

let me think through this again, if it can be cleaned in a better manner.

Reply:
I left details on all the specific callers -- "everywhere" is not all that many places, @carltongibson 's ideal can be achieved with a reasonable effort.